e2e/cypress/integration/modals/quick_switcher_spec.js

```diff
@@ -0,0 +1,143 @@
+        cy.goToDm(thirdUser.username);
+
+        // # Type either cmd+K / ctrl+K depending on OS
+        cy.get('#post_textbox').cmdOrCtrlShortcut('K');
+
+        // # Search with the term a
+        cy.focused().type('a');
+
+        // * Should have recently interacted DM on top
+        cy.get('.suggestion--selected').should('exist').and('contain.text', thirdUser.username);
+
+        // # Close quick switcher
+        cy.get('body').type('{esc}', {force: true});
+        cy.postMessage('Testing quick switcher');
+
+        // # Go to the DM channel of second user
+        cy.goToDm(secondUser.username);
+
+        cy.get('#post_textbox').cmdOrCtrlShortcut('K');
+        cy.focused().type('a');
+
+        // * Should have recently interacted DM on top
+        cy.get('.suggestion--selected').should('exist').and('contain.text', secondUser.username);
+        cy.get('body').type('{esc}', {force: true});
+    });
+
+    it('MM-T3447_3 Should match interacted users even with a partial match', () => {
+        cy.get('#post_textbox').cmdOrCtrlShortcut('K');
+
+        // # Search with the term z2
+        cy.focused().type('z2');
+
+        // Should match second user as it has a partial match with the search term
+        cy.get('.suggestion--selected').should('exist').and('contain.text', secondUser.username);
+
+        cy.get('body').type('{esc}', {force: true});
+        cy.get('#post_textbox').cmdOrCtrlShortcut('K');
+
+        // # Search with the term z3
+        cy.focused().type('z3');
+
+        // Should match third user as it has a partial match with the search term
+        cy.get('.suggestion--selected').should('exist').and('contain.text', thirdUser.username);
+        cy.get('body').type('{esc}', {force: true});
+    });
+
+    it('MM-T3447_4 Should not match GM if it is removed from LHS', () => {
+        cy.apiCreateGroupChannel([testUser.id, firstUser.id, secondUser.id]).then(({channel}) => {
+            // # Visit the newly created group message
+            cy.visit(`/${testTeam.name}/channels/${channel.name}`);
+
+            cy.get('#post_textbox').cmdOrCtrlShortcut('K');
+            cy.focused().type('a');
+
+            // * Should have recently interacted GM on top, Matching as Gaz becasue we have G prefixed for GM's
+            cy.get('.suggestion--selected').should('exist').and('contain.text', 'Gaz');
+            cy.get('body').type('{esc}', {force: true});
+
+            // # Open the channel dropdown menu
+            cy.get('#channelHeaderDropdownButton').click();
+
+            // # Click on 'Close group message' menu item
+            cy.get('#channelCloseMessage').click().wait(TIMEOUTS.HALF_SEC);
+
+            cy.get('#post_textbox').cmdOrCtrlShortcut('K');
+            cy.focused().type('a');
+
+            // * Should have recently interacted DM on top
+            cy.get('.suggestion--selected').should('exist').and('contain.text', thirdUser.username);
+        });
+    });
+});
```

Comment:
@sudheerDev thanks for adding E2E! I've ran the tests but got 3/4 failed tests - see [test report](https://mm-cypress-report.s3.amazonaws.com/1167-webapp-pr-6756-fed2c04/mochawesome.html). Also, not sure if related but would be nice to verify that the existing tests failed in this PR:
- MM-T1677 Archived channels are not shown as unread in channel switcher
- MM-T2640 - Channel appears in channel switcher before conversion but not after (for non-members of the channel)

Reply:
@saturninoabril That's wierd, that is because the first one failed i guess. Locally all 4 are passing for me. 

> MM-T1677 Archived channels are not shown as unread in channel switcher

Not valid anymore, removing the tests

> MM-T2640 - Channel appears in channel switcher before conversion but not after (for non-members of the channel)

Fixed it.

Thanks man, can you have another look at the tests added here if they are still not working for you?

Reply:
I ran the tests in CI but got all 4/4 failing - https://mm-cypress-report.s3.amazonaws.com/1174-webapp-pr-6756-92d0187/mochawesome.html. I'll try to run locally.

Reply:
They run fine for me locally
<img width="1186" alt="Untitled" src="https://user-images.githubusercontent.com/4973621/97194672-70522480-17d0-11eb-89c2-0ca47f46b46d.png">